### PR TITLE
Add tests for pure functions 

### PR DESCRIPTION
**Pull request by Hope1226**
In this feature branch I:
- Added test for updateStaus function
- Added test for updateLocalStr function 
- Added test for updateDisplay 
- Mocked DOM using JSDOM

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/updateStatus.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  const list = [
+    {
+      index: 1,
+      completed: false,
+    },
+  ];
+
+  updateComplation(list, checkbox, 0);
+  expect(list[0].completed).toBeTruthy();
+});
```
[the remark this answers is not in the document]
Comment (Hope1226) [reply src/updateStatus.test.js:23]: Hi thanks for your valuable feedback, it is highly appreciated, I have only two functions left to be tested addTask() as well as removeCompleted, I left them for the second part of this project considering their complexity. For the first part, I tried to choose as pure functions as possiable. Do you still think I need to write a test for add and delete functions? in that case I won't have anything for the second pull request.

File: src/updateDisplay.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  const page = mainPage.window.document.body;
+  const container = page.querySelector('.taskContainer');
+  const list = [{
+    index: 1,
+    description: 'some text',
+  }];
+  updateDis(list, container);
+  const tasks = container.querySelectorAll('.task');
+  expect(tasks).toHaveLength(1);
+});
```
[the remark this answers is not in the document]
Comment (henatan99) [reply src/updateDisplay.test.js]: This issue has been addressed well. Nice job! :pray:

File: src/app.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import updateLocalStr from './updateLocalStorage.js';
+import updateComplation from './updateStatus.js';
+import mainPage from './__mock__/dom.js';
+import updateDis from './updateDisplay.js';
+
+describe('Pure functions', () => {
+  test('Update display', () => {
```
Comment (henatan99) [inline src/app.test.js:7]: - [ ] It is good you add some tests for methods related to updating the todo. But as it was mentioned by the previous reviewer, testing the methods related to adding and removing to-do item are mandatory for this milestone. While what you did here are optional tests but you can still keep them. :pray: Kindly read the [project requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md) carefully.

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @Hope1226 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app.test.js (hunk 3 above)
Comment (rloterh) [reply src/app.test.js]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Ibrohim & Hope, 

Your project is complete! Great job for making the necessary changes recommended by the previous reviewer(s). There is nothing else to say other than... it's time to merge it 🍾 :shipit:.
Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have no more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**